backend/app/db/questions.py

```
@@ -555,6 +593,67 @@ async def get_question_by_id(question_id: str, user_id: str) -> Optional[Dict[st
+    also preserves the question's prior rating so feedback isn't orphaned.
+
+    The question's stale response (the answer to the old wording) is cleared on success.
+    Returns the updated question dict, or None if no document matched (already
+    regenerated concurrently, gone, or not owned).
+    """
+    questions_collection = await get_collection("questions")
+    responses_collection = await get_collection("question_responses")
+
+    try:
+        object_id = ObjectId(question_id)
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)